File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.java
class Solution {
    public int maxOperations(int[] nums, int k) {
        Arrays.sort(nums);
        int st = 0;
        int end = nums.length-1;
        int count = 0;
        while(st<end){
            int sum = nums[st]+nums[end];
            if(sum == k){
                count++;
                st++;
                end--;
            }
            else if(sum > k){
                end--;
            }
            else{
                st++;
            }
        }
        return count;
        
        
    }
}

File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.py
class Solution(object):
    def maxOperations(self, nums, k):
        nums.sort()
        st = 0
        end = len(nums)-1
        count = 0

        while st < end:
            sum = nums[st]+nums[end];
            if sum == k:
                count+=1;
                st+=1
                end-=1
            elif sum > k:
                end-=1
            else:
                st+=1
        return count;

        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        